client/fingerprint/env_digitalocean_test.go

```diff
@@ -0,0 +1,173 @@
+package fingerprint
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/nomad/client/config"
+	"github.com/hashicorp/nomad/helper/testlog"
+	"github.com/hashicorp/nomad/nomad/structs"
+)
+
+func TestDigitalOceanFingerprint_nonDigitalOcean(t *testing.T) {
+	os.Setenv("DO_ENV_URL", "http://127.0.0.1/metadata/v1/")
+	f := NewEnvDigitalOceanFingerprint(testlog.HCLogger(t))
+	node := &structs.Node{
+		Attributes: make(map[string]string),
+	}
+
+	request := &FingerprintRequest{Config: &config.Config{}, Node: node}
+	var response FingerprintResponse
+	err := f.Fingerprint(request, &response)
+	if err != nil {
+		t.Fatalf("err: %v", err)
+	}
+
+	if response.Detected {
+		t.Fatalf("expected response to not be applicable")
+	}
+
+	if len(response.Attributes) > 0 {
+		t.Fatalf("Should have zero attributes without test server")
+	}
```

Comment:
Generally we make use of `testify` to write more succinct conditions, e.g. these could boil down to 

```
require.NoError(t, err)
require.True(t, response.Detected)
require.NotEmpty(t, response.Attributes)
```